Products/ZenUI3/browser/resources/js/zenoss/ComponentPanel.js

```diff
@@ -217,16 +217,24 @@ Zenoss.nav.register({
         id: 'Edit',
         text: _t('Details'),
         action: function(node, target, combo) {
-            var uid = combo.contextUid;
-            if (!Ext.get('edit_panel')) {
-                Zenoss.form.getGeneratedForm(uid, function(config){
-                    target.add(Ext.apply({id:'edit_panel'}, config));
-                    target.layout.setActiveItem('edit_panel');
+            var uid = combo.contextUid,
+                editPanel = Ext.getCmp('edit_panel');
+            if (!editPanel) {
+                // create a static container that will handle forms;
+                editPanel = target.add({
+                    xtype: 'container',
+                    id: 'edit_panel',
+                    layout: 'fit'
                 });
-            } else {
-                target.layout.setActiveItem('edit_panel');
-                target.layout.activeItem.setContext(uid);
             }
+            target.layout.setActiveItem('edit_panel');
+            editPanel.setLoading(true);
+            Zenoss.form.getGeneratedForm(uid, function(config){
```

Comment:
we should always add new form when context changes because we receive from server bunch of form fields with data in it, if we will not remove old and add new we will always have same data in details view.